resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -438,6 +441,24 @@ private[spark] class ApplicationMaster(
     registerAM(sparkConf, rpcEnv, driverRef, sparkConf.getOption("spark.driver.appUIAddress"),
       securityMgr)
 
+    // If the credentials file config is present, we must periodically renew tokens. So create
+    // a new AMDelegationTokenRenewer
+    if (sparkConf.contains(CREDENTIALS_FILE_PATH)) {
+      // Start a short-lived thread for AMCredentialRenewer, the only purpose is to set the
+      // classloader so that main jar and secondary jars could be used by AMCredentialRenewer.
+      val credentialRenewerThread = new Thread {
+        setName("AMCredentialRenewerStarter")
+        setContextClassLoader(userClassLoader)
+
+        override def run(): Unit = {
+          startAMCredentialRenewer()
+        }
+      }
+
+      credentialRenewerThread.start()
+      credentialRenewerThread.join()
```

Comment:
Minor, but this is not capturing errors from the thread you started. It's unlikely an exception would occur at that point, but if it does, it should be re-thrown by this thread.